packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -24,18 +25,19 @@ export type MongoshCliOptions = ShellCliOptions & {
   redactInfo?: boolean;
 };
 
-export type MongoshConfigProvider = {
+export type MongoshIOProvider = {
```

Comment:
just a non-important random observation: this type is getting to be a bit of a jack of all trades, did we consider having multiple interfaces that ties together only the methods that are naturally coupled?

Reply:
Yeah, so... we could definitely split it up, but I feel like this might be one of those things where it's okay to wait until we have a reason to do so (e.g. if we ever make MongoshRepl a public API, which seems possible)

No strong feelings though :)

Reply:
Yeah, me neither, keeping it like that is super fine!